src/main/java/org/jabref/gui/fieldeditors/KeywordsEditor.java

```diff
@@ -1,24 +1,152 @@
+import javafx.scene.layout.HBox;
+
+import org.jabref.gui.ClipBoardManager;
+import org.jabref.gui.DialogService;
+import org.jabref.gui.JabRefDialogService;
+import org.jabref.gui.actions.ActionFactory;
+import org.jabref.gui.actions.SimpleCommand;
+import org.jabref.gui.actions.StandardActions;
 import org.jabref.gui.autocompleter.SuggestionProvider;
+import org.jabref.gui.keyboard.KeyBindingRepository;
+import org.jabref.gui.util.ViewModelListCellFactory;
 import org.jabref.logic.integrity.FieldCheckers;
+import org.jabref.logic.l10n.Localization;
+import org.jabref.model.entry.BibEntry;
+import org.jabref.model.entry.Keyword;
 import org.jabref.model.entry.field.Field;
 import org.jabref.preferences.PreferencesService;
 
-public class KeywordsEditor extends SimpleEditor implements FieldEditorFX {
+import com.airhacks.afterburner.views.ViewLoader;
+import com.dlsc.gemsfx.ChipView;
+import com.dlsc.gemsfx.TagsField;
+import jakarta.inject.Inject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class KeywordsEditor extends HBox implements FieldEditorFX {
+    private static final Logger LOGGER = LoggerFactory.getLogger(KeywordsEditor.class);
+
+    @FXML private TagsField<Keyword> keywordTagsField;
+
+    @Inject private PreferencesService preferencesService;
+    @Inject private DialogService dialogService;
+    @Inject private UndoManager undoManager;
+    @Inject private ClipBoardManager clipBoardManager;
+    @Inject private KeyBindingRepository keyBindingRepository;
+
+    private final KeywordsEditorViewModel viewModel;
 
     public KeywordsEditor(Field field,
                           SuggestionProvider<?> suggestionProvider,
-                          FieldCheckers fieldCheckers,
-                          PreferencesService preferences,
-                          UndoManager undoManager) {
-        super(field, suggestionProvider, fieldCheckers, preferences, undoManager);
+                          FieldCheckers fieldCheckers) {
+
+        ViewLoader.view(this)
+                  .root(this)
+                  .load();
+
+        this.viewModel = new KeywordsEditorViewModel(
+                field,
+                suggestionProvider,
+                fieldCheckers,
+                preferencesService,
+                undoManager);
+
+        keywordTagsField.setCellFactory(new ViewModelListCellFactory<Keyword>().withText(Keyword::toString));
+        keywordTagsField.setTagViewFactory(this::createTag);
+
+        keywordTagsField.setSuggestionProvider(request -> viewModel.getSuggestions(request.getUserText()));
+        keywordTagsField.setConverter(viewModel.getStringConverter());
+        keywordTagsField.setMatcher((keyword, searchText) -> keyword.get().toLowerCase().startsWith(searchText.toLowerCase()));
+        keywordTagsField.setComparator(Comparator.comparing(Keyword::get));
+
+        keywordTagsField.setNewItemProducer(searchText -> viewModel.getStringConverter().fromString(searchText));
```

Comment:
Should I remove the `ItemProducer` and the `StringConverter` since I added the requested string to the beginning of the suggestion list in `getSuggestions` method?

Reply:
I have no clue of the inner workings here. Maybe @calixtus can comment on that?

Reply:
No, that would mean that you can only add auto suggestions. what if you want to enter something, that is not suggested or (what we want to do) when the suggestion provider is turned off?

Reply:
> What if you want to enter something that is not suggested, or (what we want to do) when the suggestion provider is turned off?

I handled this in the `getSuggestions` method. Any requested string will be converted to a keyword and added at the beginning of the suggestions. This will work even if the suggestion provider is turned off and will suggest only one keyword.

https://github.com/JabRef/jabref/pull/10910/files/efb1a1400e7e123236bdff4ab08fc0b546800e29#diff-8e82ab8b0f4843d8733b604dbe23c338641c6bb6f0b6af074840a1126623554cR83-R86